planner/core/memtable_predicate_extractor.go

```diff
@@ -466,3 +475,89 @@ func (e *ClusterLogTableExtractor) Extract(
 	e.Patterns = patterns
 	return remained
 }
+
+// MetricTableExtractor is used to extract some predicates of metric_schema tables.
+type MetricTableExtractor struct {
+	extractHelper
+	// SkipRequest means the where clause always false, we don't need to request any component
+	SkipRequest bool
+	// StartTime represents the beginning time of metric data.
+	StartTime int64
+	// EndTime represents the ending time of metric data.
+	EndTime int64
+	// LabelConditions represents the label conditions of metric data.
+	LabelConditions map[string]set.StringSet
+	quantile        set.StringSet
+}
+
+// Extract implements the MemTablePredicateExtractor Extract interface
+func (e *MetricTableExtractor) Extract(
+	ctx sessionctx.Context,
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+) []expression.Expression {
+	// Extract the `quantile` columns
+	remained, skipRequest, quantile := e.extractCol(schema, names, predicates, "quantile", true)
+	e.SkipRequest = skipRequest
+	if e.SkipRequest {
+		return nil
+	}
+	e.quantile = quantile
+
+	// Extract the `time` columns
+	remained, startTime, endTime := e.extractTimeRange(ctx, schema, names, remained, "time")
```

Comment:
You can change the `extractTimeRange` and make it return time string instead of the timestamp (Or return a datum).

Reply:
Currently will be ok, the use case of extractTimeRange in the `cluster log` and `metric table` is different.